it/src/main/java/com/google/cloud/teleport/it/datastream/JDBCSource.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.datastream;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public abstract class JDBCSource {
+    private final String hostname;
+    private final String username;
+    private final String password;
+    private final int port;
+    private final Map<String, ArrayList<String>> allowedTables;
+
+    enum SourceType {
+        ORACLE,
+        MYSQL,
+        POSTGRESQL,
+    }
+
+    private final SourceType type;
+
+    public abstract SourceType getType();
+
+    JDBCSource(JDBCSourceBuilder<?> builder) {
+        this(
+                builder.hostname,
+                builder.username,
+                builder.password,
+                builder.port,
+                builder.type,
+                builder.allowedTables);
+    }
```

Comment:
I don't see a reason to have 2 constructors. I would set all the variables in this constructor and remove the other constructor